client/src/components/statCalculations/AssignDisplay.js

```diff
@@ -0,0 +1,96 @@
+import styled from "styled-components";
+import React, { Component, createRef } from "react";
+
+const Wrapper = styled.div`
+  width: 100%;
+  height: 15vh;
+  display: flex;
+  justify-content: center;
+`;
+export default class AssignDisplay extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      // statIndex tracks the index of what stat we are setting, its always one ahead
+      statIndex: 0,
+      buttonsDisabled: [false, false, false, false, false, false]
+    };
+  }
+
+  createButtons = statRolls => {
+    // temporary button array to return, one for each stat and a clear
+    let tmpButArr = [];
+
+    for (let i = 0; i < statRolls.length; i++) {
+      tmpButArr.push(
+        <button
+          key={i}
+          type="button"
+          disabled={this.state.buttonsDisabled[i]}
+          onClick={() => {
```

Comment:
Line 30. Should be it's own function instead of performing the logic within the onClick directly.

Reply:
done , handleStatAssign(statRolls,i)